Packs/PaloAltoNetworksCortex_v2/Integrations/PaloAltoNetworksCortex_v2/PaloAltoNetworksCortex_v2.py

```diff
@@ -0,0 +1,786 @@
+        Returns:
+            A query with instance_id
+        """
+        self.get_access_token()
+        integration_context = demisto.getIntegrationContext()
+        instance_id = integration_context.get(INSTANCE_ID_CONST)
+        query = re.sub(r'(?i)FROM `', f'FROM `{instance_id}.', query)
+        return query
+
+    @staticmethod
+    def generate_record_unique_id(record: Dict) -> str:
+        """
+        Since there is no unique record id and in order to add the record to the context there is a need to generte a uniqeu
+        record id
+        Args:
+            record: A query record
+
+        Returns: A unique record ID
+
+        """
+
+        def make_hashable(record_to_hash: Any) -> Any:
+            """
+            Generates a hashable version of the record so that it could be hashed.
+            Args:
+                record_to_hash: A query record that needs to become hashable
+
+            Returns:
+
+            """
+            if isinstance(record_to_hash, (tuple, list)):
+                return tuple((make_hashable(e) for e in record_to_hash))
+
+            if isinstance(record_to_hash, dict):
+                return tuple(sorted((k, make_hashable(v)) for k, v in record_to_hash.items()))
+
+            if isinstance(record_to_hash, (set, frozenset)):
+                return tuple(sorted(make_hashable(e) for e in record_to_hash))
+
+            return record_to_hash
+
+        hasher = hashlib.sha256()
+        hasher.update(repr(make_hashable(record)).encode())
+        return base64.b64encode(hasher.digest()).decode()
+
+    def add_unique_record_id(self, record):
+        record['id'] = self.generate_record_unique_id(record)
+        return record
+
+
+# TODO what to do about 'score'
+THREAT_TABLE_HEADERS = {
+    'id', 'risk_of_app', 'log_type', 'action', 'app', 'pcap_id', 'protocol', 'dest_ip', 'report_id',
+    'rule_matched', 'threat_category', 'characteristics_of_app', 'log_source_name', 'sub_type',
+    'time_generated', 'pcap', 'threat_name', 'vendor_severity', 'is_nat', 'nat_dest_port',
+    'nat_source', 'source_ip', 'app_category', 'source_location', 'dest_location', 'url_category', 'file_sha_256',
+    'file_type', 'file_name'
+}
+
+TRAFFIC_TABLE_HEADERS = {
+    'id', 'action', 'app', 'protocol', 'dest_ip', 'rule_matched', 'characteristics_of_app',
+    'log_source_name', 'risk_of_app', 'nat_source_port', 'session_start_time', 'app_sub_category', 'time_generated',
+    'is_nat', 'nat_dest_port', 'nat_dest', 'nat_source', 'source_ip', 'app_category', 'source_location', 'dest_location'
+}
+
+COMMON_HEADERS = {
+    'id', 'action', 'app', 'protocol', 'dest_ip', 'rule_matched', 'characteristics_of_app', 'log_source_name',
+    'is_nat', 'nat_dest_port', 'nat_dest', 'nat_source', 'source_ip', 'app_category', 'source_location',
+    'dest_location', 'file_type', 'file_sha_256', 'file_name', 'time_generated'
+}
+
+''' HELPER FUNCTIONS '''
+
+
+def human_readable_time_from_epoch_time(epoch_time):
```

Comment:
see if you can use `timestamp_to_datestring` from CommonServerPython instead

Reply:
not in a simple way.
`timestamp_to_datestring` formats the timestamp divided to 1e3
`human_readable_time_from_epoch_time` formats the timestamp divided to 1e6.

So I could do it only if I divide it in 1e3 and then `timestamp_to_datestring` will divide it again in 1e3.
What do you say?